apps/verticalhome/js/app.js

```diff
@@ -327,7 +403,35 @@
         // A hashchange event means that the home button was pressed.
         // The system app changes the hash of the homescreen iframe when it
         // receives a home button press.
+        case 'timeformatchange':
+          if (!this.l10nready) {
+            return;
+          }
+
+          this.timeFormat = window.navigator.mozHour12 ?
+            navigator.mozL10n.get('shortTimeFormat12') :
+            navigator.mozL10n.get('shortTimeFormat24');
+          this.refreshClock(new Date());
+
+          break;
         case 'hashchange':
+          this.backToMainScreen();
+          this.pinNavigation.reset();
+          this.clockTime.parentNode.style.opacity = 1;
```

Comment:
Setting an individual style like this seems wrong, and having to use parentNode seems wrong.  I'd define a variable to hold the parent of the clockTime element.  And I think you should define a CSS class and use that rather than hardcoding the opacity here.